### PR TITLE
First stab at a comment picture upload feature

### PR DESCRIPTION
**Pull request by KlausTrainer**

### CODE REVIEW

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 1]
```diff
@@ -25,6 +29,36 @@ app.views.CommentStream = app.views.Base.extend({
...
+
+      //debug: true,
+
+      action: "/photos",
+      params: { photo: { pending: true }},
+      allowedExtensions: this.allowedExtensions,
+      sizeLimit: this.sizeLimit,
+      messages: {
+        typeError:  Diaspora.I18n.t("photo_uploader.invalid_ext"),
+        sizeError:  Diaspora.I18n.t("photo_uploader.size_error"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Extra space before value for key 'button'.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 2]
```diff
@@ -25,6 +29,36 @@ app.views.CommentStream = app.views.Base.extend({
...
+        emptyError: Diaspora.I18n.t("photo_uploader.empty")
+      },
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Expected space or tab after '//' in comment.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 3]
```diff
@@ -25,6 +29,36 @@ app.views.CommentStream = app.views.Base.extend({
...
+      onProgress: _.bind(this.uploadProgressHandler, this),
+      onSubmit:   _.bind(this.uploadSubmitHandler, this),
+      onComplete: _.bind(this.uploadCompleteHandler, this)
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: - There should be no space after '{'
- There should be no space before '}'

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 4]
```diff
@@ -25,6 +29,36 @@ app.views.CommentStream = app.views.Base.extend({
...
+    });
+
+    this.photozoneEl.on("click", ".x", _.bind(this._removePhoto, this));
   },
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Extra space before value for key 'typeError'.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 5]
```diff
@@ -25,6 +29,36 @@ app.views.CommentStream = app.views.Base.extend({
...
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Extra space before value for key 'sizeError'.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 6]
```diff
@@ -38,9 +72,10 @@ app.views.CommentStream = app.views.Base.extend({
   createComment: function(evt) {
     if(evt){ evt.preventDefault(); }
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Extra space before value for key 'onSubmit'.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 7]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+    this.$el.append(
+      "<input type=\"hidden\" value=\""+id+"\" name=\"photos[]\" />"
+    );
+
+    // replace placeholder
+    var placeholder = this.photozoneEl.find("li.loading").first();
+        imgEl = placeholder.find("img");
+
+    placeholder
+      .removeClass("loading")
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 8]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+    this.updateTexareaValue(
+      this.textareaEl.val() + "![" +
+      Diaspora.I18n.t("publisher.markdown_editor.texts.insert_image_description_text") +
+      "](" + imgEl[0].src + " \"" +
+      Diaspora.I18n.t("publisher.markdown_editor.texts.insert_image_title") + "\")"
+    );
+  },
+
+  _cancelPhotoUpload: function() {
+    var placeholder = this.photozoneEl.find("li.loading").first();
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Multiple spaces found before '='.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 9]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+    var photo = $(evt.target).parents(".publisher_photo");
+
+    this.removePhoto(photo);
+
+    return false;
+  },
+
+  removePhoto: function(photo) {
+    photo.addClass("dim");
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 10]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+    var img = photo.find("img");
+
+    if (!img.length) {
+      photo.remove();
+      return;
+    }
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Infix operators must be spaced.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 11]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+        self = this;
+
+    $.ajax({
+      url: "/photos/" + img.attr("data-id"),
+      dataType: "json",
+      type: "DELETE",
+      success: function() {
+        $.when(photo.fadeOut(400)).then(function() {
+          photo.remove();
+          self.updateTexareaValue(self.textareaEl.val().replace(markdownRegex, ""));
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: Infix operators must be spaced.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 12]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+
+          if (self.$(".publisher_photo").length === 0) {
+            // no more photos left...
+            self.textareaWrapperEl.removeClass("with_attachments");
+          }
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: - Expected indentation of 4 space characters but found 8.
- 'imgEl' is not defined.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 13]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+
+          self.trigger("change");
+        });
+      }
+    });
+  },
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: 'imgEl' is not defined.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 14]
```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
...
+
+  // Disable submit button unless there's text or photo.
+  setSubmitButtonState: function() {
+    var onlyWhitespaces = $.trim(this.$(".comment_box").val()) === "",
+        isPhotoAttached = this.photozoneEl.children().length > 0,
+        isSubmittable = !onlyWhitespaces || isPhotoAttached;
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: 'imgEl' is not defined.

File: app/assets/javascripts/app/views/comment_stream_view.js [hunk 15]
```diff
@@ -25,6 +29,33 @@ app.views.CommentStream = app.views.Base.extend({
...
+      allowedExtensions: this.allowedExtensions,
+      sizeLimit: this.sizeLimit,
+      messages: {
+        typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
+        sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
+        emptyError: Diaspora.I18n.t("photo_uploader.empty")
+      },
+      onProgress: _.bind(this.uploadProgressHandler, this),
+      onSubmit: _.bind(this.uploadSubmitHandler, this),
+      onComplete: _.bind(this.uploadCompleteHandler, this)
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_stream_view.js]: There should be no space after '{'

**Comment by Flaburgan** [comment]
You can run pronto locally to check for coding style, see https://wiki.diasporafoundation.org/Styleguide#Automatic_local_review

**Comment by KlausTrainer** [comment]
Yeah, thanks! Just found out about that. I'm happily linting and will create a new pull request soon :)

**Comment by Flaburgan** [comment]
You didn't need to close the PR, you just to the same branch to update it. Well, you deleted the branch now so you need another PR, but for the next time... ;)

**Comment by cmrd-senya** [comment]
> you deleted the branch now so you need another PR

Not really, one may restore a deleted branch on the GitHub.

**Comment by KlausTrainer** [comment]
Thanks for letting me know!

**Pull request closed without merging**